pandas/tests/io/test_parquet.py

```diff
@@ -1179,3 +1179,13 @@ def test_close_file_handle_on_read_error(self):
                 read_parquet(path, engine="fastparquet")
             # The next line raises an error on Windows if the file is still open
             pathlib.Path(path).unlink(missing_ok=False)
+
+    def test_bytes_file_name(self, fp):
+        # GH#48944
+        df = pd.DataFrame(data={"A": [0, 1], "B": [1, 0]})
+        with tm.ensure_clean("test.parquet") as path:
+            with open(path.encode(), "wb") as f:
+                df.to_parquet(f)
```

Comment:
Could you parameterize over the engine (pyarrow and fastparquet)?

Reply:
Done

Reply:
I was originally referring to `engine` in `to_parquet`, but thanks for also add it in `read_parquet`

Reply:
Sorry, mixed that up. ``to_parquet`` does not work, because it does not support ``BufferedWriter`` as path

```
Traceback (most recent call last):
  File "/Users/patrick/Library/Application Support/JetBrains/PyCharm2022.2/scratches/scratch.py", line 272, in <module>
    df.to_parquet(f, engine="fastparquet")
  File "/Users/patrick/mambaforge/envs/random/lib/python3.10/site-packages/pandas/util/_decorators.py", line 207, in wrapper
    return func(*args, **kwargs)
  File "/Users/patrick/mambaforge/envs/random/lib/python3.10/site-packages/pandas/core/frame.py", line 2835, in to_parquet
    return to_parquet(
  File "/Users/patrick/mambaforge/envs/random/lib/python3.10/site-packages/pandas/io/parquet.py", line 420, in to_parquet
    impl.write(
  File "/Users/patrick/mambaforge/envs/random/lib/python3.10/site-packages/pandas/io/parquet.py", line 301, in write
    self.api.write(
  File "/Users/patrick/mambaforge/envs/random/lib/python3.10/site-packages/fastparquet/writer.py", line 1231, in write
    write_simple(filename, data, fmd,
  File "/Users/patrick/mambaforge/envs/random/lib/python3.10/site-packages/fastparquet/writer.py", line 880, in write_simple
    with open_with(fn, mode) as f:
TypeError: expected str, bytes or os.PathLike object, not BufferedWriter
```

Reply:
Ah okay. Guess it's a limitation in fastparquet then.